server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/RabbitMQMailQueueConfiguration.java

```diff
@@ -0,0 +1,91 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.view;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+
+public class RabbitMQMailQueueConfiguration {
+    private static final boolean DEFAULT_SIZE_METRICS_ENABLED = true;
+
+    public static class Builder {
+        private Optional<Boolean> sizeMetricsEnabled;
+
+        public Builder sizeMetricsEnabled(boolean sizeMetricsEnabled) {
+            this.sizeMetricsEnabled = Optional.of(sizeMetricsEnabled);
+            return this;
+        }
+
+        public RabbitMQMailQueueConfiguration build() {
+            return new RabbitMQMailQueueConfiguration(sizeMetricsEnabled.orElse(DEFAULT_SIZE_METRICS_ENABLED));
+        }
+    }
+
+    public static final String SIZE_METRICS_ENABLED_PROPERTY = "mailqueue.size.metricsEnabled";
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static RabbitMQMailQueueConfiguration from(Configuration configuration) {
+        return builder()
+            .sizeMetricsEnabled(configuration.getBoolean(SIZE_METRICS_ENABLED_PROPERTY, DEFAULT_SIZE_METRICS_ENABLED))
```

Comment:
We have a default value defined in two places (here and in the builder). 

This is harmful because someone willing to change it might only do it in one place and consider his job done while it's not.

To be fixing it:
 - Allow to pass an `Optional<Boolean>` as an argument of the builder
 - read the optional from the configuration

That way the builder only is handling the default value.